staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -603,7 +642,10 @@ func CopyConfig(config *Config) *Config {
 		},
 		AuthProvider:        config.AuthProvider,
 		AuthConfigPersister: config.AuthConfigPersister,
-		ExecProvider:        config.ExecProvider,
+		Exec: Exec{
+			ExecProvider: config.Exec.ExecProvider,
+			Config:       config.Exec.Config,
```

Comment:
should this deep copy the config.Exec.Config if it is non-nil?

Reply:
I think that sounds reasonable...

Reply:
https://github.com/kubernetes/kubernetes/pull/95489/commits/bd4af0df52b7da2a838795ccf9a8760a0316da48

Reply:
BTW, I started to add a test to verify that we deep copy this member, but then backed off since it seemed ugly and I didn't know if we needed it. I can add it back if you think that would be healthier. It was something to the tune of this.
```go
		if actual.ExecProvider != nil {
			if expected.ExecProvider == nil {
				t.Fatal("CopyConfig dropped ExecProvider field")
			}
			if actual.ExecProvider.Config != nil {
				if expected.ExecProvider.Config == nil {
					t.Fatal("CopyConfig dropped ExecProvider.Config field")
				}
				if actual.ExecProvider.Config `is the object same as` expected.ExecProvider.Config {
					t.Fatalf("CopyConfig did not deep copy ExecProvider.Config field (%p == %p)", actual.ExecProvider.Config, expected.ExecProvider.Config)
				}
			}
		}

```